Roll engine to version dd6f46c485192f4506035088c9065b9f5dbba9ab

a-siva opened the pull request:
Includes:
* Make flutter_test on Mac exit on error like Linux and Windows. (flutter/engine#4873)
* Revert "Fix vulkan include path (flutter/engine#4870)" (#4872)
* Allow FirstFrameListeners to remove themselves from the FlutterView's list (flutter/engine#4871)
* Fix vulkan include path (flutter/engine#4870)
* Support for decimal and signed numeric keyboard (flutter/engine#4853)
* CONTRIBUTING.md: explain the need for building host with non-host (flutter/engine#4863
* Roll Dart to 2f68e82526d59b679a522e25db224d8c7e23d280 (flutter/engine#4864)
* Remove declaration for undefined GPUSurfaceGL::SelectPixelConfig (flutter/engine#4857)
* Vulkan swap chain setup no longer uses GrPixelConfig and doesn't incl…
* Revert dart roll 20180322 (flutter/engine#4855)
* Roll back pub upgrade. (flutter/engine#4854)
* Roll flutter 20180322a (flutter/engine#4852)
* Document Window.defaultRouteName a little better. (flutter/engine#4851)
* Roll dart to 95e9e890a9 (flutter/engine#4850)
* Fix GPUSurfaceGL includes on Linux (flutter/engine#4849)
* Slightly improve the docs for the internal _futurize method (flutter/engine#4847)
* Fix supported color type check on iOS simulators (flutter/engine#4846)
* Fix GPUSurfaceGL includes on macOS (flutter/engine#4844)
* Handle Apple-specific GLES headers (flutter/engine#4843)
* [fuchsia] Fix up includes of Zircon headers (flutter/engine#4842)
* VulkanSurface and GPUSurfaceGL no longer use GrPixelConfig (flutter/engine#4814)
* Roll Skia (flutter/engine#4841)
* Provide asset lookup key on ios (flutter/engine#4817)
* Load kernel files from the script URI in the isolate create callback (flutter/engine#4828)
* Remove some unneeded profiling flags (flutter/engine#4827)
* Remove the canvas save/restore calls in Paragraph::Paint (flutter/engine#4822)

a-siva committed:
First roll engine version to 3e877d371a359097d7af28aa6e85026fa2318baa
the version just before the roll of Dart into the engine.

a-siva committed:
Roll engine version to e622bd93c902a308788c28f3637325154b9e9cc4
which contains a roll of Dart in it.

a-siva committed:
Merge branch 'master' of github.com:flutter/flutter into rollengine

aam commented:
Looks like few tests are failing
```
⏩ RUNNING: cd packages/flutter_driver; ../../bin/flutter test
00:00 +0: /home/travis/build/flutter/flutter sdk/packages/flutter_driver/test/flutter_driver_test.dart: FlutterDriver.connect connects to isolate paused at start
00:00 +0 -1: loading /home/travis/build/flutter/flutter sdk/packages/flutter_driver/test/src/extension_test.dart
00:00 +0 -1: /home/travis/build/flutter/flutter sdk/packages/flutter_driver/test/flutter_driver_test.dart: FlutterDriver.connect connects to isolate paused at start [E]
  type 'MockVM' is not a subtype of type 'Future<VM>' where
    MockVM is from file:///home/travis/build/flutter/flutter%20sdk/packages/flutter_driver/test/flutter_driver_test.dart
    Future is from dart:async
    VM is from package:vm_service_client/src/vm.dart
  
  package:vm_service_client/vm_service_client.dart      MockVMServiceClient.getVM
  package:flutter_driver/src/driver/driver.dart 165:32  FlutterDriver.connect
```

a-siva committed:
Fix some strong mode errors.

a-siva committed:
Fix analyzer warning.

a-siva committed:
Correctly fix analyzer warning.

a-siva committed:
Fix more strong mode errors resulting from the use of the mockito
package.

a-siva committed:
Updated engine version to dd6f46c485192f4506035088c9065b9f5dbba9ab
in order to pick up the fix for engine exiting when an error is
encountered reading a dart script.
